pkg/util/httprequest/httprequest.go

```diff
@@ -100,8 +85,6 @@ func SchemeHost(req *http.Request) (string /*scheme*/, string /*host*/) {
 		scheme = "https"
 	case len(req.URL.Scheme) > 0:
 		scheme = req.URL.Scheme
-	case port == "443":
```

Comment:
we're guessing it either way. in the absence of proto info, using https for port 443 is more sane